### ISSUE TITLE
question about missing dependency wheel files (shexjsg depends on antlr)

### ISSUE THREAD

**Issue by sideshowtom** [linkml/linkml#1072]
Not a python user, but I'm just trying to do enough to use linkml-runtime in a browser with javascript.  I'm using Pyodide and micropip.install and it comes up with an error about missing wheel files in pyldmod and others:

oops: PythonError: Traceback (most recent call last):
  File "/lib/python3.9/asyncio/futures.py", line 201, in result
    raise self._exception
  File "/lib/python3.9/asyncio/tasks.py", line 258, in __step
    result = coro.throw(exc)
  File "/lib/python3.9/site-packages/_pyodide/_base.py", line 494, in eval_code_async
    await CodeRunner(
  File "/lib/python3.9/site-packages/_pyodide/_base.py", line 347, in run_async
    await coroutine
  File "<exec>", line 7, in <module>
  File "/lib/python3.9/asyncio/futures.py", line 284, in __await__
    yield self  # This tells Task to wait for completion.
  File "/lib/python3.9/asyncio/tasks.py", line 328, in __wakeup
    future.result()
  File "/lib/python3.9/asyncio/futures.py", line 201, in result
    raise self._exception
  File "/lib/python3.9/asyncio/tasks.py", line 256, in __step
    result = coro.send(None)
  File "/lib/python3.9/site-packages/micropip/_micropip.py", line 191, in install
    raise ValueError(
ValueError: Couldn't find a pure Python 3 wheel for: 'pyldmod', 'antlr4-python3-runtime~=4.9', 'pyldmod', 'pyyaml<6.0.0,>=5.1.0', 'wrapt<2,>=1.10'

**Comment by sierra-moxon** [comment]
@pkalita-lbl - could you comment here on the feasibility/status of this ticket?  Should we revisit anything?  Should we close it?

**Comment by pkalita-lbl** [comment]
I think we've gone backwards since the last time we looked at this. In the current stable [Pyodide terminal](https://pyodide.org/en/stable/console.html):

```python
>>> import micropip
>>> micropip.__version__
'0.2.0'
>>> await micropip.install('linkml-runtime==1.4.5')
Traceback (most recent call last):
  File "<console>", line 1, in <module>
  . . . 
ValueError: Can't find a pure Python 3 wheel for 'pytrie'.
```

Indeed, `pytrie` only publishes a [source distribution](https://pypi.org/project/PyTrie/0.4.0/#files) -- no wheels. We don't use `pytrie` directly; it's brought in through `curies`:

```
➜ poetry show pytrie --why --tree
curies 0.4.0 Idiomatic conversion between URIs and compact URIs (CURIEs).
└── pytrie *
```

It's @cmungall's call at the end of the day, but my two cents is that even if we got this going it would be very susceptible to breaking any time we touched `linkml-runtime`'s dependencies. I think investing the time in https://github.com/linkml/linkml-runtime.js (which has had new functionality added since it was first mentioned in this thread!) might be better in the long run.
